pilot/pkg/networking/apigen/apigen_test.go

```diff
@@ -32,7 +32,7 @@ import (
 )
 
 var (
-	grpcAddr         = "127.0.0.1:14056"
+	grpcAddr         = "127.0.0.1:14057"
```

Comment:
I didn't get a chance to dig into the full code path but generally if you set port to 0 it will auto assign an open port. Is that feasible here?

Reply:
I think if it auto assigns a port I cant get the port to call from the test, we would need a callback that returns the port.

Reply:
I looked at the portselector thing and that would work, ill implement it.

Reply:
you might be able to, depends where this is used. Sorry on my phone so cannot look closer. But normally if it's used in net.Listen() or similar you can then later get the address of it. We do it in a number of tests

Reply:
ds.GRPCListener.Address?

Reply:
The port selector is pretty heavyweight and not actually correct (you cannot reserve a port without actually binding to it, so there is a race)

Reply:
Yeah portselector seems pretty heavyweight. 

I think the `:0` port could also work, but i am about to commit something, check it out to see if it might work. Also, Wouldnt we have to extract the port from the string `ds.GRPCListener.Address()`? Seems a bit messy but can be done.

Reply:
Yep it fails. going with `:0`.

Reply:
I actually am not sure we even need the port anywhere.. I think the other usage of 14056 is just a coincidence but I could be wrong

Reply:
https://github.com/istio/istio/blob/master/pilot/pkg/networking/grpcgen/grpcgen_test.go#L43

https://github.com/istio/istio/blob/master/pilot/pkg/networking/grpcgen/grpcgen_test.go#L46

https://github.com/istio/istio/blob/master/pilot/pkg/networking/grpcgen/grpcgen_test.go#L58

https://github.com/istio/istio/blob/master/pilot/pkg/networking/grpcgen/grpcgen_test.go#L62

https://github.com/istio/istio/blob/master/pilot/pkg/networking/grpcgen/grpcgen_test.go#L85

https://github.com/istio/istio/blob/master/pilot/pkg/networking/grpcgen/grpcgen_test.go#L139

and in an EXTERNAL JSON:

https://github.com/istio/istio/blob/master/pilot/pkg/networking/grpcgen/testdata/xds_bootstrap.json#L4

I honestly think bumping up the port is the way to go from here, unless we have to overwrite the json file and replace all the instances.

Reply:
I think the env var `GRPC_XDS_BOOTSTRAP` is being set to the json file, and something is reading a port from there.

Reply:
yep all of those need to be changed. this is complex.

Reply:
I'm just going to go with bumping the port. objections? 

the issue is that before we init the grpc service we need the port to add http services to the `ds.MemRegistry`.

Reply:
bumped port everywhere

Reply:
ah shoot, was looking at the wrong file. sorry @howardjohn those references dont count